datasetinsights/datasets/transformers/coco.py

```diff
@@ -0,0 +1,144 @@
+        self._bbox2d_definition_id = bbox2d_definition_id
+        self._captures = Captures(
+            data_root=data_root, version=const.DEFAULT_PERCEPTION_VERSION
+        ).filter(def_id=bbox2d_definition_id)
+        self._annotation_definitions = AnnotationDefinitions(
+            data_root, version=const.DEFAULT_PERCEPTION_VERSION
+        )
+
+    def execute(self, output):
+        """Execute COCO Transformer
+
+        Args:
+            output (str): the output directory where converted dataset will
+              be stored.
+        """
+        self._copy_images(output)
+        self._process_instances(output)
+
+    def _copy_images(self, output):
+        image_to_folder = Path(output) / "images"
+        image_to_folder.mkdir(parents=True, exist_ok=True)
+        for _, row in self._captures.iterrows():
+            image_from = self._data_root / row["filename"]
+            if not image_from.exists():
+                continue
+            capture_id = uuid_to_int(row["id"])
+            image_to = image_to_folder / f"camera_{capture_id}.png"
+            shutil.copy(str(image_from), str(image_to))
+
+    def _process_instances(self, output):
+        output = Path(output) / "annotations"
+        output.mkdir(parents=True, exist_ok=True)
+        instances = {
+            "info": {"description": "COCO compatible Synthetic Dataset"},
+            "licences": [{"url": "", "id": 1, "name": "default"}],
+            "images": self._images(),
+            "annotations": self._annotations(),
+            "categories": self._categories(),
+        }
+        output_file = output / "instances.json"
+        with open(output_file, "w") as out:
+            json.dump(instances, out)
+
+    def _images(self):
+        images = []
+        for _, row in self._captures.iterrows():
+            image_file = self._data_root / row["filename"]
+            if not image_file.exists():
+                continue
+            with Image.open(image_file) as im:
+                width, height = im.size
+            capture_id = uuid_to_int(row["id"])
+            record = {
+                "license": 1,
+                "file_name": f"camera_{capture_id}.png",
+                "coco_url": "",
+                "height": height,
+                "width": width,
+                "date_captured": "",
+                "flickr_url": "",
+                "id": capture_id,
+            }
+            images.append(record)
+
+        return images
+
+    def _annotations(self):
+        annotations = []
+        for _, row in self._captures.iterrows():
+            image_id = uuid_to_int(row["id"])
+            for ann in row["annotation.values"]:
+                x = ann["x"]
+                y = ann["y"]
+                w = ann["width"]
+                h = ann["height"]
+                area = float(w) * float(h)
+                record = {
+                    "segmentation": [],  # TODO: parse instance segmentation map
```

Comment:
Is this going to be implemented later?

Reply:
Not in this PR. It will be tackled [separately](https://github.com/Unity-Technologies/datasetinsights/issues/159)